### PR TITLE
Add WithCommandLoggingDisabled to otelmongo tracer

### PR DESCRIPTION
**Pull request by davidgwcurve**
Add `WithCommandLoggingDisabled` to otelmongo tracer so the user can opt out of tracing the mongo statements. This is useful for applications which may log sensitive data but still want mongo tracing.

Fixes Issue #712

### CODE REVIEW

**Comment by davidgwcurve** [comment]
hey @dashpole @jmacd @MrAlias @evantorrie @XSAM, this is available for review now with all checks passed. I had issues with getting the CLA signed, EasyCLA was playing up. Please let me know if you have any comments.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,11 +8,13 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
-## [0.19.0] - 2021-03-19
+## [0.19.0] - 2021-03-29
```
Comment (MrAlias) [inline CHANGELOG.md:11]: This should not change, it describes the point in time this release was made.

```suggestion
## [0.19.0] - 2021-03-19
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,11 +8,13 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
 
 ### Changed
 
 - Upgrade to v0.19.0 of `go.opentelemetry.io/otel`.
+- The `go.opentelemetry.io/contrib/instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo` instrumentation now accepts a `WithCommandLoggingDisabled`, 
+   so the caller can specify whether to opt-out of tracing the mongo command. (#712)
```
Comment (MrAlias) [inline CHANGELOG.md:17]: This needs to be moved to the unreleased section (a new `### Changed` section needs to be added there.

File: Makefile [hunk 3]
```diff
@@ -22,7 +22,7 @@ endif
 
 
 GOTEST_MIN = go test -v -timeout 30s
-GOTEST = $(GOTEST_MIN) -race
+GOTEST = CGO_ENABLED=1 $(GOTEST_MIN) -race
```
Comment (MrAlias) [inline Makefile:25]: Why is this added? This will apply to all packages tested. If it is actually needed we need to isolate it to where it actually is needed.

File: instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/config.go [hunk 4]
```diff
@@ -55,3 +57,11 @@ func WithTracerProvider(provider trace.TracerProvider) Option {
 		cfg.TracerProvider = provider
 	}
 }
+
+// WithCommandLoggingDisabled specifies if tracing of the mongo command is disabled or not.
+// If none is specified, logging of the mongo command is enabled.
```
Comment (MrAlias) [inline instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/config.go:62]:
```suggestion
// WithCommandLoggingDisabled specifies if tracing of the MongoDB command is disabled or not.
// If none is specified, logging of the MongoDB command is enabled.
```

File: instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/config.go [hunk 5]
```diff
@@ -55,3 +57,11 @@ func WithTracerProvider(provider trace.TracerProvider) Option {
...
+func WithCommandLoggingDisabled(disabled bool) Option {
```
Comment (MrAlias) [inline instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/config.go:63]: I'm against using the term `Logging` here. This term will describe the logging signal that we plan to add to the project at a later date.

```suggestion
func WithCommandAttributeDisabled(disabled bool) Option {
```

File: CHANGELOG.md (hunk 2 above)
Comment (davidgwcurve) [reply CHANGELOG.md:17]: sure thing, thanks

File: Makefile (hunk 3 above)
Comment (davidgwcurve) [reply Makefile:25]: this was because I ran `make` and I got an error saying that `CGO_ENABLED` needed to be enabled to run `-race`. I've just enabled it locally instead.

File: instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/config.go [hunk 6]
```diff
@@ -55,3 +57,11 @@ func WithTracerProvider(provider trace.TracerProvider) Option {
 		cfg.TracerProvider = provider
 	}
 }
+
+// WithCommandAttributeDisabled specifies if tracing of the MongoDB command is disabled or not.
+// If none is specified, tracing of the MongoDB command is enabled.
```
Comment (MrAlias) [inline instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/config.go:62]:
```suggestion
// WithCommandAttributeDisabled specifies if the MongoDB command is added as an attribute to Spans or not.
// The MongoDB command will be added as an attribute to Spans by default if this option is not provided.
```

File: Makefile (hunk 3 above)
Comment (MrAlias) [reply Makefile:25]: I'm still seeing this change. Are you still planning to push a revert here?

Comment (davidgwcurve) [reply Makefile]: 🤦 , thank you!

**Comment by davidgwcurve** [comment]
thank you @MrAlias for the review!

**Review by MrAlias — APPROVED** [review]

**Comment by davidgwcurve** [comment]
what is the process for getting this merged @MrAlias ?

**Comment by MrAlias** [comment]
> what is the process for getting this merged @MrAlias ?

[Another review from @open-telemetry/go-approvers or @open-telemetry/go-maintainers is required](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/CONTRIBUTING.md#how-to-get-prs-merged)

**Review by XSAM — APPROVED** [review]
Sorry for the delay. It looks good.
